crates/database/src/fee_policies.rs

```diff
@@ -0,0 +1,114 @@
+use {
+    crate::{auction::AuctionId, OrderUid, PgTransaction},
+    sqlx::PgConnection,
+    std::ops::DerefMut,
+};
+
+#[derive(Debug, Clone, PartialEq, sqlx::FromRow)]
+pub struct FeePolicyRow {
+    pub auction_id: AuctionId,
+    pub order_uid: OrderUid,
+    pub kind: FeePolicyKindRow,
+    pub price_improvement_factor: Option<f64>,
+    pub max_volume_factor: Option<f64>,
+    pub volume_factor: Option<f64>,
+}
+
+#[derive(Debug, Clone, PartialEq, sqlx::Type)]
+#[sqlx(type_name = "PolicyKind", rename_all = "lowercase")]
+pub enum FeePolicyKindRow {
+    PriceImprovement,
+    Volume,
+}
+
+pub async fn insert(
+    ex: &mut PgTransaction<'_>,
```

Comment:
You could take a `PgConnection` here since `PgTransaction` automatically dereferences into that type. Unless you want to enforce that you can only use this function in a transaction.

Reply:
Fixed